File: python/src/unittest/QuesoChihuahuaTest.py
import unittest
from QuesoChihuahua import *

class QuesoChihuahuaTest(unittest.TestCase):
    def setUp(self):
        self.quesoChihuahua = QuesoChihuahua()

    def test_CurrentTemperature(self):
        self.quesoChihuahua.setCurrentTemperature(21)
        self.assertEquals(21, self.quesoChihuahua.getCurrentTemperature())

    def test_CurrentTemperature(self):
        self.quesoChihuahua.setCurrentTemperature(21)
        self.assertEquals(21, self.quesoChihuahua.getCurrentTemperature())

    def test_FalseMelt(self):
        self.quesoChihuahua.melt(False)
        self.assertFalse(self.quesoChihuahua.isMelted())

    def test_TrueMelt(self):
        self.quesoChihuahua.melt(True)
        self.assertTrue(self.quesoChihuahua.isMelted())

    def test_Melting(self):
        self.assertEquals(20, self.quesoChihuahua.getMeltingTemperature())

File: csharp/unittest-practice/src/test/QuesoChihuahuaTest.cs
using NUnit.Framework;
using unittestpractice.main;

namespace unittestpractice.test
{
    [TestFixture]
    internal class QuesoChihuahuaTest
    {
        private QuesoChihuahua _quesoChihuahua;

        [SetUp]
        public void Setup()
        {
            _quesoChihuahua = new QuesoChihuahua();
        }

        [TestCase]
        public void TestCurrentTemperature()
        {
            _quesoChihuahua.SetCurrentTemperature(21);
            Assert.AreEqual(21, _quesoChihuahua.GetCurrentTemperature());
        }

        [TestCase]
        public void TestFalseMelt()
        {
            _quesoChihuahua.Melt(false);
            Assert.IsFalse(_quesoChihuahua.IsMelted());
        }

        [TestCase]
        public void TestTrueMelt()
        {
            _quesoChihuahua.Melt(true);
            Assert.IsTrue(_quesoChihuahua.IsMelted());
        }

        [TestCase]
        public void TestMelting()
        {
            Assert.AreEqual(20, _quesoChihuahua.GetMeltingTemperature());
        }
    }
}